### ISSUE TITLE
Can't build on OSX

### ISSUE THREAD

**Issue by mpaw** [ReservedField/evic-sdk#17]
I'm having a problem getting the examples built on OSX. I installed all the dependencies and set all the env variables.

```
➜  helloworld git:(master) ✗ echo $ARMGCC
/Users/marcin/Development/evic/gcc-arm-none-eabi-5.2
➜  helloworld git:(master) ✗ echo $EVICSDK
/Users/marcin/Development/evic/evic-sdk
```

Doing make in the root directory of the project looks to complete successfully. But trying make from within an examples folder errors out:

```
➜  helloworld git:(master) ✗ make
mkdir -p bin
arm-none-eabi-ld main.o -L/usr/arm-none-eabi/lib -L/usr/lib/arm-none-eabi/newlib -L/usr/lib/gcc/arm-none-eabi/5.2.1 -L/Users/marcin/Development/evic/evic-sdk/lib -nostdlib -nostartfiles -T/Users/marcin/Development/evic/evic-sdk/linker/linker.ld --gc-sections -o bin/helloworld.elf
arm-none-eabi-ld: cannot find armv7e-m/libgcc.a
arm-none-eabi-ld: cannot find armv7e-m/libc.a
arm-none-eabi-ld: cannot find armv7e-m/libm.a
make: *** [helloworld.elf] Error 1
```

The files are there though:

```
➜  armv7e-m pwd
/Users/marcin/Development/evic/gcc-arm-none-eabi-5.2/arm-none-eabi/lib/armv7e-m
➜  armv7e-m ls
aprofile-validation.specs libnosys.a                nosys.specs
aprofile-ve.specs         librdimon.a               pid.specs
cpu-init                  librdimon_nano.a          rdimon-crt0.o
crt0.o                    librdpmon.a               rdimon.specs
fpu                       libstdc++.a               rdpmon-crt0.o
iq80310.specs             libstdc++.a-gdb.py        rdpmon.specs
libc.a                    libstdc++_nano.a          redboot-crt0.o
libc_nano.a               libsupc++.a               redboot-syscalls.o
libg.a                    libsupc++_nano.a          redboot.ld
libg_nano.a               linux-crt0.o              redboot.specs
libgloss-linux.a          linux.specs               softfp
libm.a                    nano.specs
```

Has anyone gotten this working on OSX?

**Comment by Ban3** [comment]
$ARMGCC is only used on Windows, it defaults to /usr on other systems. That's why you have `/usr/lib/gcc/arm-none-eabi/5.2.1`as a library path instead of the correct one. @ReservedField should $ARMGCC overriding work on every OS?

**Comment by ReservedField** [comment]
Fixed in 1b7ad47.
@mpaw Do you confirm?

**Comment by mpaw** [comment]
Yep, that fixed it! Thanks

**Comment by mpaw** [comment]
Just took a look at the README update. Wanted to point something out. I tried many ways to get `arm-none-eabi` working with this repo. The brew recipes that install it (I found a few distinct ones) all end up giving the same error when you try to make the examples. The libraries must not be getting installed into the same paths as Linux.

I ended up falling back on the package built by the arm team:
https://launchpad.net/gcc-arm-embedded

If there is indeed a way to go about this without having to set `ARMGCC` I would be interested to hear it. I'm sure lowering the difficulty of getting a development system set up can only help this SDK.

**Comment by ReservedField** [comment]
I implemented automatic toolchain detection in 176d936. Remove the `ARMGCC` var and try it out, it's tested on Windows and Linux but I'd like to know if it works on OSX. I used a cd/pwd hack instead of readlink since I read `readlink -f` doesn't work on OSX, so it _should_ work.

**Comment by mpaw** [comment]
Back to the same error:

```
➜  helloworld git:(master) ✗ make
arm-none-eabi-gcc -Wall -mcpu=cortex-m4 -mthumb -Os -fdata-sections -ffunction-sections -I/Users/marcin/Development/evic/evic-sdk/nuvoton-sdk/Library/CMSIS/Include -I/Users/marcin/Development/evic/evic-sdk/nuvoton-sdk/Library/Device/Nuvoton/M451Series/Include -I/Users/marcin/Development/evic/evic-sdk/nuvoton-sdk/Library/StdDriver/inc -I/Users/marcin/Development/evic/evic-sdk/include -c main.c -o main.o
mkdir -p bin
arm-none-eabi-ld main.o -L//arm-none-eabi/lib -L//lib/arm-none-eabi/newlib -L//lib/gcc/arm-none-eabi/5.2.1 -L/Users/marcin/Development/evic/evic-sdk/lib -nostdlib -nostartfiles -T/Users/marcin/Development/evic/evic-sdk/linker/linker.ld --gc-sections -o bin/helloworld.elf
arm-none-eabi-ld: cannot find armv7e-m/libgcc.a
arm-none-eabi-ld: cannot find armv7e-m/libc.a
arm-none-eabi-ld: cannot find armv7e-m/libm.a
make: *** [helloworld.elf] Error 1
```

What assumptions are made about where the library should be?

**Comment by ReservedField** [comment]
Basically no assumptions, GCC is asked for the paths and then they are parsed.
Could you post the output of `arm-none-eabi-gcc --print-search-dir`?

**Comment by mpaw** [comment]
Here you go

```
➜  Development arm-none-eabi-gcc --print-search-dir
install: /usr/local/Cellar/gcc-arm-none-eabi/20151219/bin/../lib/gcc/arm-none-eabi/5.2.1/
programs: =/usr/local/Cellar/gcc-arm-none-eabi/20151219/bin/../lib/gcc/arm-none-eabi/5.2.1/:/usr/local/Cellar/gcc-arm-none-eabi/20151219/bin/../lib/gcc/:/usr/local/Cellar/gcc-arm-none-eabi/20151219/bin/../lib/gcc/arm-none-eabi/5.2.1/../../../../arm-none-eabi/bin/arm-none-eabi/5.2.1/:/usr/local/Cellar/gcc-arm-none-eabi/20151219/bin/../lib/gcc/arm-none-eabi/5.2.1/../../../../arm-none-eabi/bin/
libraries: =/usr/local/Cellar/gcc-arm-none-eabi/20151219/bin/../lib/gcc/arm-none-eabi/5.2.1/:/usr/local/Cellar/gcc-arm-none-eabi/20151219/bin/../lib/gcc/:/usr/local/Cellar/gcc-arm-none-eabi/20151219/bin/../lib/gcc/arm-none-eabi/5.2.1/../../../../arm-none-eabi/lib/arm-none-eabi/5.2.1/:/usr/local/Cellar/gcc-arm-none-eabi/20151219/bin/../lib/gcc/arm-none-eabi/5.2.1/../../../../arm-none-eabi/lib/:/usr/local/Cellar/gcc-arm-none-eabi/20151219/bin/../arm-none-eabi/lib/arm-none-eabi/5.2.1/:/usr/local/Cellar/gcc-arm-none-eabi/20151219/bin/../arm-none-eabi/lib/:/usr/local/Cellar/gcc-arm-none-eabi/20151219/bin/../arm-none-eabi/usr/lib/arm-none-eabi/5.2.1/:/usr/local/Cellar/gcc-arm-none-eabi/20151219/bin/../arm-none-eabi/usr/lib/
```

**Comment by mpaw** [comment]
I just checked and it looks like `ARMGCC` gets set to `/` in the Makefile

**Comment by mpaw** [comment]
I found the problem and fixed it. There are quirks with `sed` on OSX that don't happen on Windows/Linux. More specifically the filter was failing and doing very weird things and the newline wasn't getting detected. Instead of the `:` getting replaced by a newline it was getting replaced with literal `n`. The workarounds I found are not reliable and it's much easier to just do this a different way on OSX.

I put in a PR (#18) with this fix. I also added directions for how to set up the toolchain. The brew recipes I found were all outdated so I put one up that has the latest version.

Everything is working properly now with these updates.
